test/TabExpansion.Tests.ps1

```diff
@@ -0,0 +1,38 @@
+. $PSScriptRoot\Shared.ps1
+
+Describe 'TabExpansion Tests' {
+    Context 'Fetch/Push/Pull TabExpansion Tests' {
+        It 'Tab completes all remotes' {
+            $result = & $module GitTabExpansionInternal 'git push '
+            $result | Should BeExactly 'origin'
+        }
+        It 'Tab completes matching remotes' {
+            $result = & $module GitTabExpansionInternal 'git push or'
+            $result | Should BeExactly 'origin'
+        }
+        It 'Tab completes remote and all branches' {
+            $result = & $module GitTabExpansionInternal 'git push origin '
+            $result -contains 'master' | Should Be $true
+            $result -contains 'origin/master' | Should Be $true
+            $result -contains 'origin/HEAD' | Should Be $true
+        }
+        It 'Tab completes remote and matching branches' {
+            $result = & $module GitTabExpansionInternal 'git push origin ma'
+            $result | Should BeExactly 'master'
+        }
+        It 'Tab completes matching remote with preceding parameters' {
+            $result = & $module GitTabExpansionInternal 'git push --follow-tags  -u   or'
+            $result | Should BeExactly 'origin'
+        }
+        It 'Tab completes remote and all branches with preceding parameters' {
+            $result = & $module GitTabExpansionInternal 'git push  --follow-tags  -u   origin '
+            $result -contains 'master' | Should Be $true
+            $result -contains 'origin/master' | Should Be $true
+            $result -contains 'origin/HEAD' | Should Be $true
+        }
+        It 'Tab completes remote and matching branch with preceding parameters' {
+            $result = & $module GitTabExpansionInternal 'git push  --follow-tags  -u   origin ma'
+            $result | Should BeExactly 'master'
+        }
+    }

```

Comment:
Add tests to assert empty result when expanding a missing remote, branch, or remote/branch combination?

It wouldn't be a terrible idea to set a remote and/or branch name with a `-` in it, as a sanity check for the `gitParams` regex.